### PR TITLE
This turns butchering into something that resembles a component.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

See title. Does that by not having it be placed in the middle of item_attack.d and removing a bunch of getComponents.

## Why It's Good For The Game

Old thing was shit. This is also shit but slightly less. The mech drill and the recycler do really silly things with this component.

## Changelog
:cl:
refactor: Butchering component refactored
balance: Edaggers are now sharp if they get turned on.
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Unrelated, but #37723 is still open and it really shouldn't be. Worked when I tested it.

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -150,6 +150,7 @@
 #define COMSIG_MOB_HUD_CREATED "mob_hud_created"				//from base of mob/create_mob_hud(): ()
 #define COMSIG_MOB_ATTACK_HAND "mob_attack_hand"				//from base of
 #define COMSIG_MOB_ITEM_ATTACK "mob_item_attack"				//from base of /obj/item/attack(): (mob/M, mob/user)
+	#define COMPONENT_ITEM_NO_ATTACK 1
```
Comment (actioninja) [inline code/__DEFINES/components.dm]: why the tab in front?

**Review by actioninja — CHANGES_REQUESTED** [review]

File: code/__DEFINES/components.dm (hunk 1 above)
Comment (nemvar) [reply code/__DEFINES/components.dm]: For readability. COMPONENT_ITEM_NO_ATTACK is basically a possible return value of COMSIG_MOB_ITEM_ATTACK. This is the standard way of writing this.

Comment (actioninja) [reply code/__DEFINES/components.dm]: Sounds fair.

**Review by actioninja — DISMISSED** [review]

**Comment by actioninja** [comment]
conflict

**Pull request merged**